hive-plugins/src/main/java/co/cask/hydrator/plugin/batch/commons/HiveSchemaConverter.java

```diff
@@ -57,14 +58,15 @@ public static HCatSchema toHiveSchema(Schema schema, HCatSchema tableSchema) {
         Preconditions.checkNotNull(hCatFieldSchema, "Missing field %s in table schema", name);
         PrimitiveTypeInfo hiveType = hCatFieldSchema.getTypeInfo();
         PrimitiveTypeInfo type = getType(name, field.getSchema());
-        if (hiveType != type) {
+        if (!hiveType.equals(type)) {
           LOG.warn("The given schema {} for the field {} does not match the schema {} from the table. " +
-                     "The schema {} for field {} be used.", type, name, hiveType, hiveType, name);
+                     "The schema {} for field {} will be used.", type, name, hiveType, hiveType, name);
         }
         fields.add(hCatFieldSchema);
       } catch (HCatException e) {
         LOG.error("Failed to create HCatFieldSchema field {} of type {} from schema", name,
-                  field.getSchema().getType());
+                  field.getSchema().getType(), e);
+        Throwables.propagate(e);
```

Comment:
I think we should just throw an exception with the message that is being logged, and with e as its parent instead of logging and throwing a RuntimeException. Perhaps as an IllegalArgumentException since the caller is, sending in bad input?

Though it's kind of a moot point. Looking at the code, tableSchema.get(name) declares that it throws HCatException, but actually will never throw an HCatException. Good job Hive...